gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/SnappySelectResultSet.java

```diff
@@ -192,6 +194,11 @@ public ExecRow getNextRow() throws StandardException {
       this.setCurrentRow(this.currentRow);
       return this.currentRow;
     } catch(Exception ex) {
+      try {
+        Misc.checkIfCacheClosing(ex);
+      } catch(CacheClosedException cacheCloseException) {
+        throw cacheCloseException;
+      }
```

Comment:
Why this additional catch and re-throw? Just `Misc.checkIfCacheClosing(ex)` should be enough without the additional try-catch block.